### PR TITLE
Eng 8713

### PR DESCRIPTION
**Pull request by akhanzode**
Drain pending transactions. Additional logging. Close offset manager which was leaking.

### CODE REVIEW

**Comment by rbetts** [comment]
Looks like this also resolves eng-8736.

**Comment by akhanzode** [comment]
@rbetts Yes...I did

**Comment by rbetts** [comment]
(thanks)

File: src/frontend/org/voltdb/importclient/kafka/KafkaStreamImporter.java [hunk 1]
```diff
@@ -671,6 +677,23 @@ public void run() {
                             }
                         } catch (Exception ex) {
                             error(ex, "Failed to fetch from %s", m_topicAndPartition);
+                            //See if its network error and find new leader for this partition.
+                            if (ex instanceof ClosedChannelException) {
```
Comment (vtkstef) [inline src/frontend/org/voltdb/importclient/kafka/KafkaStreamImporter.java]: would we not be better off with a different catch block?

**Comment by vtkstef** [comment]
R.by: moi

**Comment by akhanzode** [comment]
squash merged. r=vtkstef

**Pull request closed without merging**